src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -2239,6 +2239,138 @@ class DropExtensionStatementSegment(BaseSegment):
     )
 
 
+class AlterExtensionStatementSegment(BaseSegment):
+    """An `ALTER EXTENSION` statement.
+
+    https://www.postgresql.org/docs/16/sql-alterextension.html
+    """
+
+    type = "alter_extension_statement"
+    match_grammar: Matchable = Sequence(
+        "ALTER",
+        "EXTENSION",
+        Ref("ExtensionReferenceSegment"),
+        OneOf(
+            Sequence(
+                "UPDATE",
+                Sequence(
+                    "TO",
+                    Ref("LiteralGrammar"),
+                    optional=True,
+                ),
+            ),
+            Sequence(
+                "SET",
+                "SCHEMA",
+                OneOf(Ref("SchemaReferenceSegment"), "CURRENT_SCHEMA"),
+            ),
+            Sequence(
+                OneOf(
+                    "ADD",
+                    "DROP",
+                ),
+                OneOf(
+                    Sequence(
+                        OneOf(
+                            Sequence("ACCESS", "METHOD"),
+                            "COLLATION",
+                            "CONVERSION",
+                            "DOMAIN",
+                            Sequence("EVENT", "TRIGGER"),
+                            Sequence("FOREIGN", "DATA", "WRAPPER"),
+                            Sequence("FOREIGN", "TABLE"),
+                            Sequence(
+                                Ref.keyword("PROCEDURAL", optional=True),
+                                "LANGUAGE",
+                            ),
+                            "SCHEMA",
+                            "SEQUENCE",
+                            "SERVER",
+                            Sequence(
+                                "TEXT",
+                                "SEARCH",
+                                OneOf(
+                                    "CONFIGURATION",
+                                    "DICTIONARY",
+                                    "PARSER",
+                                    "TEMPLATE",
+                                ),
+                            ),
+                            "TYPE",
+                        ),
+                        Ref("ObjectReferenceSegment"),
+                    ),
+                    Sequence(
+                        OneOf(
+                            Sequence("MATERIALIZED", "VIEW"),
+                            "TABLE",
+                            "VIEW",
+                        ),
+                        Ref("TableReferenceSegment"),
+                    ),
+                    Sequence(
+                        "AGGREGATE",
+                        Ref("ObjectReferenceSegment"),
+                        Bracketed(
+                            Sequence(
+                                # TODO: Is this too permissive?
+                                Anything(),
+                                optional=True,
+                            ),
+                            optional=True,
+                        ),
```

Comment:
This snippet is stolen from `CommentOnStatementSegment` as it references the same convoluted `aggregate_signature`.

For reference, from PG documentation:
```
aggregate_signature is:

* |
[ argmode ] [ argname ] argtype [ , ... ] |
[ [ argmode ] [ argname ] argtype [ , ... ] ] ORDER BY [ argmode ] [ argname ] argtype [ , ... ]
```